File: src/main/java/com/ddd/trainings/proposal/domain/TrainingTemplate.java
package com.ddd.trainings.proposal.domain;

import com.ddd.trainings.idea.domain.IdeaId;
import com.ddd.utils.BaseEntity;
import lombok.Getter;

public class TrainingTemplate extends BaseEntity {

  @Getter
  private final TemplateId id;
  private final IdeaId ideaId;
  @Getter
  private final String review;

  public TrainingTemplate(TemplateId id, IdeaId ideaId, String review) {
    this.id = id;
    this.ideaId = ideaId;
    this.review = review;
  }

  @Override
  public TemplateId getId() {
    return id;
  }
} 

File: src/Ddd/Trainings/Proposal/Domain/TrainingTemplate.cs
using Ddd.Trainings.Idea.Domain;
using Ddd.Utils;

namespace Ddd.Trainings.Proposal.Domain
{
    public class TrainingTemplate : BaseEntity<TemplateId>
    {
        public TemplateId Id { get; }
        public string Name { get; }
        public string Description { get; }
        public string[] Keywords { get; }

        public TrainingTemplate(TemplateId id, string name, string description, string[] keywords)
        {
            Id = id;
            Name = name;
            Description = description;
            Keywords = keywords;
        }

        public override TemplateId GetId()
        {
            return Id;
        }
    }
} 